metals/src/main/scala/scala/meta/internal/metals/ConnectionProvider.scala

```diff
@@ -681,10 +682,44 @@ class ConnectionProvider(
                     buildToolProvider.optProjectRoot
                   )
                 ) {
-                  // TODO(olafur) try to connect but gracefully error
-                  languageClient.showMessage(
-                    Messages.ImportProjectPartiallyFailed
-                  )
+                  buildToolProvider.optProjectRoot match {
+                    case None =>
+                      languageClient.showMessage(
+                        Messages.ImportProjectPartiallyFailed
+                          .params(canShowLogs = false)
+                      )
+                    case Some(projectRoot) =>
+                      languageClient
+                        .showMessageRequest(
+                          Messages.ImportProjectPartiallyFailed
+                            .params(canShowLogs = true)
+                        )
+                        .asScala
+                        .foreach {
+                          case Messages.ImportProjectPartiallyFailed.showLogs =>
+                            val logsFile = projectRoot.resolve(Directories.log)
+                            val logsPath = logsFile.toURI.toString
+                            val linesCount =
+                              logsFile.readText.linesIterator.size
+                            val cursorRange = new lsp4j.Range(
+                              new lsp4j.Position(linesCount, 0),
+                              new lsp4j.Position(linesCount, 0),
+                            )
+                            val location =
+                              new lsp4j.Location(logsPath, cursorRange)
+                            languageClient.metalsExecuteClientCommand(
+                              ClientCommands.GotoLocation
```

Comment:
Would that be better to somehow use the existing `Go to log` command?

https://github.com/scalameta/metals/blob/e7f290cda1b72f6c07d985d62b4d6eb26e513dce/metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala#L1030-L1049